spec/system/consumer/split_checkout_tax_incl_spec.rb

```diff
@@ -0,0 +1,226 @@
+          login_as(user_within_zone)
+        end
+
+        it "will be charged tax on the order" do
+          visit checkout_step_path(:details)
+
+          choose "Delivery"
+
+          click_button "Next - Payment method"
+          click_on "Next - Order summary"
+          click_on "Complete order"
+
+          # UI checks
+          expect(page).to have_content("Confirmed")
+          expect(page).to have_selector('#order_total', text: with_currency(10.00))
+          expect(page).to have_selector('#tax-row', text: with_currency(1.15))
+
+          # DB checks
+          assert_db_tax_incl
+        end
+
+        after { logout }
+      end
+    end
+
+    describe "for a customer with shipping address outside the tax zone" do
+      context "on legacy checkout" do
+        before do
+          set_order order_outside_zone
+          login_as(user_outside_zone)
+        end
+
+        it "will not be charged tax on the order" do
+          pending("#7540")
+          visit checkout_path
+
+          find(:xpath, '//*[@id="shipping"]/ng-form/dd').click
+          choose free_shipping.name.to_s
+
+          within "#payment" do
+            choose free_payment.name.to_s
+          end
+
+          click_on "Place order now"
+
+          # UI checks
+          expect(page).to have_selector('#order_total', text: with_currency(10.00))
+          expect(page).not_to have_content("includes tax")
+
+          # DB checks
+          assert_db_no_tax_incl
+        end
+
+        after { logout }
+      end
+
+      context "on split-checkout" do
+        before do
+          allow(Flipper).to receive(:enabled?).with(:split_checkout).and_return(true)
+          allow(Flipper).to receive(:enabled?).with(:split_checkout, anything).and_return(true)
+
+          set_order order_outside_zone
+          login_as(user_outside_zone)
+        end
+
+        it "will not be charged tax on the order" do
+          pending("#7540")
+          visit checkout_step_path(:details)
+
+          choose "Delivery"
+          check "order_save_bill_address"
+          check "ship_address_same_as_billing"
+
+          click_button "Next - Payment method"
+          click_on "Next - Order summary"
+          click_on "Complete order"
+
+          # UI checks
+          expect(page).to have_content("Confirmed")
+          expect(page).to have_selector('#order_total', text: with_currency(10.00))
+          expect(page).not_to have_content("includes tax")
+
+          # DB checks
+          assert_db_no_tax_incl
+        end
+
+        after { logout }
+      end
+    end
+  end
+end
+
+private
+
+def assert_db_tax_incl
+  order_within_zone.reload
+  expect(order_within_zone.included_tax_total).to eq(1.15)
+  expect(order_within_zone.additional_tax_total).to eq(0.0)
+end
+
+def assert_db_no_tax_incl
+  order_outside_zone.reload
+  expect(order_outside_zone.included_tax_total).to eq(0.0)
+  expect(order_outside_zone.additional_tax_total).to eq(0.0)
+end
```

Comment:
I just noticed that these methods are outside of the `describe` block! They are defined globally when the spec file is loaded. I'll change that.

Reply:
I guess that could also be a reason for the (previous) flakyness? :shrug: 
Anyway, thanks addressing that :+1: